pkg/terraform/providers.go

```diff
@@ -1,33 +1,38 @@
 package terraform
 
 import (
-	"github.com/hashicorp/go-plugin"
 	"github.com/sirupsen/logrus"
 )
 
 const (
 	AWS string = "aws"
 )
 
-var providers = make(map[string]TerraformProvider)
+type ProviderLibrary struct {
+	providers map[string]TerraformProvider
+}
 
-func AddProvider(name string, provider TerraformProvider) {
-	providers[name] = provider
+func NewLibrary() *ProviderLibrary {
```

Comment:
`NewProviderLibrary` ?

Reply:
+1 also here `terraform.NewLibrary` is not missleading but not really clear either